common/schema/definition.go

```diff
@@ -228,69 +243,113 @@ func flows() Schema {
+				ClickHouseGenerateFrom: fmt.Sprintf("dictGetOrDefault('%s', 'region', SrcAddr, '')", DictionaryNetworks),
 			},
 			{
 				Key:                    ColumnDstNetRegion,
 				ParserType:             "string",
 				ClickHouseType:         "LowCardinality(String)",
-				ClickHouseGenerateFrom: "dictGetOrDefault('networks', 'region', DstAddr, '')",
+				ClickHouseGenerateFrom: fmt.Sprintf("dictGetOrDefault('%s', 'region', DstAddr, '')", DictionaryNetworks),
 			},
 			{
 				Key:                    ColumnSrcNetTenant,
 				ParserType:             "string",
 				ClickHouseType:         "LowCardinality(String)",
-				ClickHouseGenerateFrom: "dictGetOrDefault('networks', 'tenant', SrcAddr, '')",
+				ClickHouseGenerateFrom: fmt.Sprintf("dictGetOrDefault('%s', 'tenant', SrcAddr, '')", DictionaryNetworks),
 			},
 			{
 				Key:                    ColumnDstNetTenant,
 				ParserType:             "string",
 				ClickHouseType:         "LowCardinality(String)",
-				ClickHouseGenerateFrom: "dictGetOrDefault('networks', 'tenant', DstAddr, '')",
+				ClickHouseGenerateFrom: fmt.Sprintf("dictGetOrDefault('%s', 'tenant', DstAddr, '')", DictionaryNetworks),
 			},
 			{Key: ColumnSrcVlan, ParserType: "uint", ClickHouseType: "UInt16", Disabled: true, Group: ColumnGroupL2},
-			{Key: ColumnSrcCountry, ParserType: "string", ClickHouseType: "FixedString(2)"},
+			{
+				Key:                    ColumnSrcCountry,
+				ParserType:             "string",
+				ClickHouseType:         "FixedString(2)",
+				ClickHouseGenerateFrom: fmt.Sprintf("dictGetOrDefault('%s', 'country', SrcAddr, '')", DictionaryNetworks),
+			},
+			{
+				Key:                    ColumnDstCountry,
+				ParserType:             "string",
+				ClickHouseType:         "FixedString(2)",
+				ClickHouseGenerateFrom: fmt.Sprintf("dictGetOrDefault('%s', 'country', DstAddr, '')", DictionaryNetworks),
+			},
+			{
+				Key:                    ColumnSrcGeoCity,
+				ParserType:             "string",
+				ClickHouseType:         "LowCardinality(String)",
+				ClickHouseGenerateFrom: fmt.Sprintf("dictGetOrDefault('%s', 'city', SrcAddr, '')", DictionaryNetworks),
+			},
+			{
+				Key:                    ColumnDstGeoCity,
+				ParserType:             "string",
+				ClickHouseType:         "LowCardinality(String)",
+				ClickHouseGenerateFrom: fmt.Sprintf("dictGetOrDefault('%s', 'city', DstAddr, '')", DictionaryNetworks),
+			},
+			{
+				Key:                    ColumnSrcGeoState,
+				ParserType:             "string",
+				ClickHouseType:         "LowCardinality(String)",
+				ClickHouseGenerateFrom: fmt.Sprintf("dictGetOrDefault('%s', 'state', SrcAddr, '')", DictionaryNetworks),
+			},
+			{
+				Key:                    ColumnDstGeoState,
+				ParserType:             "string",
+				ClickHouseType:         "LowCardinality(String)",
+				ClickHouseGenerateFrom: fmt.Sprintf("dictGetOrDefault('%s', 'state', DstAddr, '')", DictionaryNetworks),
+			},
```

Comment:
State and City should be optional. Otherwise, people will pay the lookup cost even when there is no database loaded for that.

Reply:
do you mean optional as they can activate the column on demand via config ?

Should it be linked to whether or not the geoip enrich is configured or not ?